libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -1134,14 +1134,58 @@ bool NavEKF2::getHeightControlLimit(float &height) const
     return core[primary].getHeightControlLimit(height);
 }
 
-// return the amount of yaw angle change due to the last yaw angle reset in radians
+// return the amount of yaw angle change (in radians) due to the last yaw angle reset or core selection switch
 // returns the time of the last yaw angle reset or 0 if no reset has ever occurred
-uint32_t NavEKF2::getLastYawResetAngle(float &yawAng) const
+uint32_t NavEKF2::getLastYawResetAngle(float &yawAngDelta)
 {
     if (!core) {
         return 0;
     }
-    return core[primary].getLastYawResetAngle(yawAng);
+
+    // check for an internal ekf yaw reset
+    float temp_yaw_delta;
+    uint32_t ekf_reset_ms = core[primary].getLastYawResetAngle(temp_yaw_delta);
+    if (ekf_reset_ms != yaw_step_data.last_ekf_reset_ms) {
+        // record the time of the ekf's internal yaw reset event
+        yaw_step_data.last_ekf_reset_ms = ekf_reset_ms;
+
+        // record the the ekf's internal yaw reset value
+        yaw_step_data.yaw_delta = temp_yaw_delta;
+
+        // record the yaw reset event time
+        yaw_step_data.yaw_reset_time_ms = imuSampleTime_us/1000;
+
+    }
+
+    // check for a core switch and if a switch has occurred, set the yaw reset delta
+    // to the difference in yaw angle between the current and last yaw angle
+    Vector3f eulers_primary;
+    core[primary].getEulerAngles(eulers_primary);
+    if (primary != yaw_step_data.prev_instance) {
+        // the delta is the difference between the current and previous yaw
+        // This overwrites any yaw reset value recorded from an internal eff reset
+        yaw_step_data.yaw_delta += wrap_PI(eulers_primary.z - yaw_step_data.prev_yaw);
```

Comment:
The comment says it overwrites any yaw reset value but this is doing an addition, should it just be an equal?